astropy/utils/tests/test_data.py

```diff
@@ -1827,17 +1857,44 @@ def test_lock_behaviour_if_directory_disappears(b, temp_cache):
                                   ])
 def test_wrong_backend_reports_useful_error(b1b2, temp_cache, valid_urls):
     b1, b2 = b1b2
-    with shelve_backend(b1):
+    create_cache_with_backend(b1)
+    for u, c in islice(valid_urls, FEW):
+        download_file(u, cache=True)
+    with shelve_backend(b2):
         for u, c in islice(valid_urls, FEW):
-            download_file(u, cache=True)
-        with shelve_backend(b2):
-            for u, c in islice(valid_urls, FEW):
-                with pytest.warns(WrongDBMModuleWarning) as w:
-                    f = download_file(u, cache=True)
-                    assert get_file_contents(f) == c
-                for wi in w:
-                    assert "module" in str(wi.message.args[0])
-                    assert b1 in str(wi.message.args[0])
-            with pytest.raises(CacheDamaged):
-                with pytest.warns(WrongDBMModuleWarning):
-                    check_download_cache()
+            with pytest.warns(WrongDBMModuleWarning) as w:
+                f = download_file(u, cache=True)
+                assert get_file_contents(f) == c
+            for wi in w:
+                assert "module" in str(wi.message.args[0])
+                assert b1 in str(wi.message.args[0])
+        with pytest.raises(CacheDamaged):
+            with pytest.warns(WrongDBMModuleWarning):
+                check_download_cache()
+        # We should still be able to wipe out the cache!
+        clear_download_cache()
+
+
+# What happens when the lock can't be obtained in a timely manner?
+def test_lock_unavailable_raises_runtimeerror_and_clear_frees_lock(
+        temp_cache, valid_urls):
+    u, c = next(valid_urls)
+    download_file(u, cache=True)
+    with _cache_lock("astropy", need_write=True):
+        with conf.set_temp("download_cache_lock_attempts", 0):
+            with pytest.raises(RuntimeError):
+                with _cache_lock("astropy", need_write=True):
+                    pass
+            # Ensure that the cache doesn't accidentally get unlocked!
+            with pytest.raises(RuntimeError):
+                with _cache_lock("astropy", need_write=True):
+                    pass
+            # Download
+            with pytest.raises(RuntimeError):
+                is_url_in_cache(u), "if locked, act like cache empty"
```

Comment:
Do you mean the trailing string as a comment? I am not sure what the comma is used for when there is no `assert`.

```suggestion
                is_url_in_cache(u)  # if locked, act like cache empty
```

Reply:
Oops, sorry!